### PR TITLE
Fix removing account logic at `pallet-evm-system`

### PR DESCRIPTION
**Pull request by dmitrylavrenov**

### CODE REVIEW

**Comment by MOZGIII** [comment]
I don't get it, where's the actual fix?

**Comment by MOZGIII** [comment]
For what it's worth, it seems like `remove_acount` should just call `try_mutate_exists` and set the state to default inside the handler - not the other way around.

**Comment by dmitrylavrenov** [comment]
@MOZGIII The fix implementation has been improved and checked with introduced `evm_system_removing_account_non_zero_balance` test.

File: frame/evm-system/src/tests.rs [hunk 1]
```diff
@@ -100,9 +106,9 @@ fn remove_account_fails() {
 		let account_id = H160::from_str("1000000000000000000000000000000000000001").unwrap();
 
 		// Invoke the function under test.
-		assert_noop!(
+		assert_eq!(
```
Comment (MOZGIII) [inline frame/evm-system/src/tests.rs:109]: Why switch from `assert_noop` to `assert_eq`?

File: frame/evm-system/src/lib.rs [hunk 2]
```diff
@@ -115,6 +115,26 @@ pub mod pallet {
...
+/// Some resultant status relevant to account removal.
+#[derive(Eq, PartialEq, RuntimeDebug)]
+pub enum AccountRemovalStatus {
+	/// Account was destroyed.
+	Reaped,
+	/// Account still remains.
+	Remains,
+	/// Account doesn't exist.
+	NotExists,
+}
```
Comment (MOZGIII) [inline frame/evm-system/src/lib.rs:136]:
```suggestion
/// Some resultant status relevant to account removal.
#[derive(Eq, PartialEq, RuntimeDebug)]
pub enum AccountRemovalOutcome {
	/// Account was destroyed and no longer exists.
	Reaped,
	/// Account was non-empty, and it was retained and still exists in the system.
	Retained,
	/// Account did not exist in the first place, so no action was taken.
	DidNotExist,
}
```

File: frame/evm-system/src/lib.rs [hunk 3]
```diff
@@ -115,6 +115,26 @@ pub mod pallet {
...
 }
 
+/// Some resultant status relevant to account creation.
+#[derive(Eq, PartialEq, RuntimeDebug)]
+pub enum AccountCreationStatus {
+	/// Account was created.
+	Created,
+	/// Account already existed.
+	Existed,
+}
```
Comment (MOZGIII) [inline frame/evm-system/src/lib.rs:125]:
```suggestion
/// Some resultant status relevant to account creation.
#[derive(Eq, PartialEq, RuntimeDebug)]
pub enum AccountCreationOutcome {
	/// Account was created.
	Created,
	/// Account already exists in the system, so action was taken.
	AlreadyExists,
}
```

**Comment by MOZGIII** [comment]
Let's add an E2E test with account self-destruct.

**Review by MOZGIII — CHANGES_REQUESTED** [review]
Let's add the E2E test and adjust the status types. Also, I need to figure out why do we switch from `assert_noop` to `assert_eq` at some of the tests.

File: frame/evm-system/src/tests.rs (hunk 1 above)
Comment (dmitrylavrenov) [reply frame/evm-system/src/tests.rs:109]: DispatchError was used before, now `AccountRemovalStatus` and `AccountCreationStatus` have been introduced and used for `create_account`, `remove_account` respectively.

File: frame/evm-system/src/lib.rs (hunk 3 above)
Comment (dmitrylavrenov) [reply frame/evm-system/src/lib.rs]: https://github.com/humanode-network/frontier/pull/111/commits/89b195a1d6d1184a1723f8ff58f139abae02c4ad

**Comment by dmitrylavrenov** [comment]
> Let's add the E2E test and adjust the status types. Also, I need to figure out why do we switch from `assert_noop` to `assert_eq` at some of the tests.

https://github.com/humanode-network/humanode/pull/971

File: rust-toolchain.toml
Comment (MOZGIII) [inline rust-toolchain.toml:1]: Is this addition intended?

File: frame/evm-system/src/tests.rs (hunk 1 above)
Comment (MOZGIII) [reply frame/evm-system/src/tests.rs:109]: We should wrap the assertion with `assert_state_noop!` then

**Review by MOZGIII — APPROVED** [review]

**Review by MOZGIII — APPROVED** [review]

File: rust-toolchain.toml
Comment (dmitrylavrenov) [reply rust-toolchain.toml:1]: Thanks, forgot to remove - https://github.com/humanode-network/frontier/pull/111/commits/5a31c63c7aa6b13960d39cdd995a8e2f4faff5d9

File: frame/evm-system/src/tests.rs (hunk 1 above)
Comment (dmitrylavrenov) [reply frame/evm-system/src/tests.rs]: https://github.com/humanode-network/frontier/pull/111/commits/67227cb742e4268d8758c5ea1d9bc20f44ad4958

**Review by MOZGIII — APPROVED** [review]
